common/src/ProductInfo.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Azure.Devices.Common
+{
+    internal class ProductInfo
```

Comment:
Missing test. Given the current implementation for DeviceClient, I'm not sure this actually works as expected.